### PR TITLE
feat(debugging): expand devtools documentation, add debugging in VSCode, closes #628

### LINKED ISSUE #628: [docs] Guide for debugging Rust (ex. with LLDB via VS Code)
It would be really nice to be able to debug the Rust code in a Tauri app. I've done it before with pure Rust projects, but I'm very fuzzy on how I would attach to a Tauri process. The current docs only describe how to use `println!`-based debugging.

### PR DESCRIPTION
**Pull request by lucasfernog**

### CODE REVIEW

File: docs/guides/08-debugging/application.md
Comment (kasper9n) [inline docs/guides/08-debugging/application.md]: `Cmd+Option+I` doesn't work on macOS, at least on my OS version

File: docs/guides/08-debugging/application.md
Comment (lorenzolewis) [inline docs/guides/08-debugging/application.md]:
```suggestion
or like this on Windows:
```

File: docs/guides/08-debugging/application.md
Comment (kasper9n) [inline docs/guides/08-debugging/application.md]:
```suggestion
The Core process is powered by Rust so you can use GDB or LLDB to debug it. You can follow the [Debugging in VSCode] guide to learn how to use the LLDB VSCode Extension to debug the Core Process of Tauri applications.
```

File: docs/guides/08-debugging/application.md
Comment (lorenzolewis) [inline docs/guides/08-debugging/application.md:45]: A quick screenshot here of the actual right-click menu would be a good visual maybe

Comment (lorenzolewis) [reply docs/guides/08-debugging/application.md:45]: Also note that which web inspector depends on the OS (macOS being safari's, etc.)

File: docs/guides/08-debugging/application.md
Comment (lorenzolewis) [inline docs/guides/08-debugging/application.md]:
```suggestion
### Opening Devtools Programatically
```

File: docs/guides/08-debugging/application.md
Comment (lorenzolewis) [inline docs/guides/08-debugging/application.md]:
```suggestion
You can also use the `Ctrl + Shift + i` shortcut on Linux and Windows, and `Command + Option + i` on macOS to open the inspector.
```

File: docs/guides/08-debugging/application.md
Comment (lorenzolewis) [inline docs/guides/08-debugging/application.md]: Add links to the API docs for these

File: docs/guides/08-debugging/application.md
Comment (lorenzolewis) [inline docs/guides/08-debugging/application.md]:
```suggestion
### Using the Inspector in Production
```

File: docs/guides/08-debugging/application.md
Comment (lorenzolewis) [reply docs/guides/08-debugging/application.md]: I was also going to ask about this, it's been a bit since I've tested but I couldn't get this to work before (can test with a fresh install of Tauri if you'd like)

File: docs/guides/08-debugging/application.md
Comment (lorenzolewis) [inline docs/guides/08-debugging/application.md]:
```suggestion
By default the inspector is only enabled in development and debug builds unless you enable it with a Cargo feature.
```

File: docs/guides/08-debugging/application.md
Comment (lorenzolewis) [inline docs/guides/08-debugging/application.md]: Are this and the below bullet steps to doing this? If so, I'd make them headings (h4) with `1. ` and `2. ` at the beginning

File: docs/guides/08-debugging/application.md
Comment (lorenzolewis) [inline docs/guides/08-debugging/application.md]:
```suggestion
You can also run a built app from the terminal, giving you the Rust compiler notes (in case of errors) or your `println` messages. Browse to the file `src-tauri/target/(release|debug)/[app name]` and either double-click it (note: the terminal closes on errors) or just run it in directly in your console.
```

File: docs/guides/08-debugging/application.md
Comment (lorenzolewis) [inline docs/guides/08-debugging/application.md]:
```suggestion
The devtools API is private on macOS. Using private APIs on macOS prevents your application from being accepted for the App Store.
```

Comment (lorenzolewis) [reply docs/guides/08-debugging/application.md]: If there's a link to a reference on this their explains why (or just generically why you shouldn't enable private APIs for App Store builds), it would be good to include here.

File: docs/guides/08-debugging/application.md
Comment (lorenzolewis) [inline docs/guides/08-debugging/application.md]:
```suggestion
The Core process is powered by Rust so you can use GDB or LLDB to debug it. You can follow the [Debugging in VS Code] to learn how to use the LLDB VS Code Extension to debug the Core Process of Tauri applications.
```

File: docs/guides/08-debugging/application.md
Comment (lorenzolewis) [inline docs/guides/08-debugging/application.md]:
```suggestion
[Debugging in VS Code]: vs-code
```

File: docs/guides/08-debugging/vscode.md
Comment (lorenzolewis) [inline docs/guides/08-debugging/vscode.md]:
```suggestion
# Debugging in VS Code
```

Comment (lorenzolewis) [inline docs/guides/08-debugging/vscode.md]: Filename should also be `vs-code.md`

File: docs/guides/08-debugging/vscode.md
Comment (lorenzolewis) [inline docs/guides/08-debugging/vscode.md]:
```suggestion
This guide describes how to setup debugging in VS Code for the [Core Process in Tauri applications](../architecture/process-model#the-core-process).
```

Comment (lorenzolewis) [reply docs/guides/08-debugging/vscode.md]: Not 100% sure that the MD link format is correct above ^

File: docs/guides/08-debugging/vscode.md
Comment (lorenzolewis) [inline docs/guides/08-debugging/vscode.md]:
```suggestion
Create a `.vscode/launch.json` file and paste the below JSON contents into it:
```

File: docs/guides/08-debugging/vscode.md
Comment (lorenzolewis) [inline docs/guides/08-debugging/vscode.md]:
```suggestion
Note that it does not use the Tauri CLI, so exclusive CLI features are not executed. The `beforeDevCommand` and `beforeBuildCommand` scripts must be executed beforehand or configured as a task in the `preLaunchTask` field. Below is an example `.vscode/tasks.json` file that has two tasks, one for a `beforeDevCommand` that spawns a development server and one for `beforeBuildCommand`:
```

File: docs/guides/08-debugging/vscode.md
Comment (lorenzolewis) [inline docs/guides/08-debugging/vscode.md]:
```suggestion
Now you can set breakpoints in `src-tauri/src/main.rs` or any other Rust file and start debugging by pressing `F5`.
```

**Comment by lorenzolewis** [comment]
Also, not sure if your title is correct for which issue this is closing since this PR is #632 (so technically it would close itself 😉)

File: docs/guides/08-debugging/application.md
Comment (lorenzolewis) [reply docs/guides/08-debugging/application.md]: Actually, I take it back, working for me on macOS 12.4, Tauri-api rc 6 and Tauri-cli rc 13

Comment (kasper9n) [reply docs/guides/08-debugging/application.md]: Must be an OS version thing then, it's not working for me on macOS 10.15

**Comment by lucasfernog** [comment]
Oops totally confused the issue number lol

File: docs/guides/08-debugging/application.md
Comment (lucasfernog) [reply docs/guides/08-debugging/application.md]: I think the wry implementation only uses a newer API (the devtools API is private so apple can change it a lot). please file an issue on the wry repo @probablykasper I think I can set up a VM and try to fix it

Comment (kasper9n) [reply docs/guides/08-debugging/application.md]: Isn't the `Cmd+Opt+I` shortcut provided by the webview itself? `window.open_devtools()` works fine, it's just the shortcut that doesn't work

Comment (lucasfernog) [reply docs/guides/08-debugging/application.md]: it's not provided by the webview, we're doing it

Comment (kasper9n) [reply docs/guides/08-debugging/application.md]: ah, nice

File: docs/guides/08-debugging/application.md
Comment (lucasfernog) [reply docs/guides/08-debugging/application.md:45]: one screenshot for each OS? with tabs?

Comment (lorenzolewis) [reply docs/guides/08-debugging/application.md:45]: Could possibly do that, I can grab one of macOS and put it here (and can put the tabs together too)

Comment (lorenzolewis) [reply docs/guides/08-debugging/application.md:45]: Screenshot 2022-05-30 at 09 53 16

Something like this? It could also be that this is overkill looking at it now ;)

Comment (lucasfernog) [reply docs/guides/08-debugging/application.md:45]: seems like i'm too dumb to make several screenshots the same size and menu position :D

Comment (lorenzolewis) [reply docs/guides/08-debugging/application.md:45]: Honestly, don't worry about it, this specific place probably doesn't make sense. And always easy to add them in later if we decide we want to later.

Comment (lucasfernog) [reply docs/guides/08-debugging/application.md:45]: ok PR done then

**Comment by lorenzolewis** [comment]
Just going to go through and do a few formatting suggestions for the new stuff

File: docs/guides/08-debugging/application.md
Comment (lorenzolewis) [reply docs/guides/08-debugging/application.md:45]: I think it still makes sense to have a quick blurb about the console being dependent on the OS's default browser (unless I missed that being added in)

Comment (lucasfernog) [reply docs/guides/08-debugging/application.md:45]: yeah i'll add that

File: docs/guides/08-debugging/application.md
Comment (lorenzolewis) [inline docs/guides/08-debugging/application.md]:
```suggestion
#### Create a Debug Build
```
Bold and such doesn't show up in headings anyway

File: docs/guides/08-debugging/application.md
Comment (lorenzolewis) [inline docs/guides/08-debugging/application.md]:
```suggestion
You can also run a built app from the terminal, giving you the Rust compiler notes (in case of errors) or your `println` messages. Browse to the file `src-tauri/target/(release|debug)/[app name]` and run it in directly in your console or double-click the executable itself in the filesystem (note: the console closes on errors with this method).
```

File: docs/guides/08-debugging/application.md
Comment (lorenzolewis) [inline docs/guides/08-debugging/application.md]:
```suggestion
#### Enable Devtools Feature
```

File: docs/guides/08-debugging/application.md
Comment (lorenzolewis) [inline docs/guides/08-debugging/application.md]:
```suggestion

:::warning
The devtools API is private on macOS. Using private APIs on macOS prevents your application from being accepted for the App Store.
:::

To enable the devtools in production builds, you must enable the `devtools` Cargo feature in the `src-tauri/Cargo.toml` file:

```toml
[dependencies]
tauri = { version = "...", features = ["...", "devtools"] }
```
```

**Review by lorenzolewis — COMMENTED** [review]
Just a few minor tweaks

**Comment by lorenzolewis** [comment]
Alright, hopefully my last review of being picky ;)

**Comment by lorenzolewis** [comment]
I think this makes sense to have this show right under "Development" in the sidebar, but I'll do that change in https://github.com/tauri-apps/tauri-docs/pull/631 once this is merged since I'm removing the folder prefixes there anyway

**Comment by lucasfernog** [comment]
I agree with you, but I didn't want to change the structure too much here.

**Review by lorenzolewis — APPROVED** [review]
Looks brilliant, will leave it to you to merge.

**Comment by lucasfernog** [comment]
Thanks for the help!

**Comment by lorenzolewis** [comment]
lol, slipping in that API change under the radar

**Pull request merged**
